snark-verifier-sdk/src/aggregation.rs

```diff
@@ -52,12 +55,103 @@ pub fn flatten_accumulator<'a>(
         .truncation
         .limbs
         .into_iter()
-        .chain(lhs.y.truncation.limbs.into_iter())
-        .chain(rhs.x.truncation.limbs.into_iter())
-        .chain(rhs.y.truncation.limbs.into_iter())
+        .chain(lhs.y.truncation.limbs)
+        .chain(rhs.x.truncation.limbs)
+        .chain(rhs.y.truncation.limbs)
         .collect()
 }
 
+type AssignedInstances<'a> =
+    Vec<Vec<<BaseFieldEccChip as EccInstructions<'a, G1Affine>>::AssignedScalar>>;
+type KzgAcc<'a> = KzgAccumulator<G1Affine, Rc<Halo2Loader<'a>>>;
+type AssignedEcPoints<'a> =
+    Vec<Vec<<EccChip<Fr, FpConfig<Fr, Fq>> as EccInstructions<'a, G1Affine>>::AssignedEcPoint>>;
+type AssignedScalars<'a> =
+    Vec<Option<<EccChip<Fr, FpConfig<Fr, Fq>> as EccInstructions<'a, G1Affine>>::AssignedScalar>>;
+
+/// Core function used in `synthesize` to aggregate multiple `snarks`.
+///  
+/// Returns the assigned instances of previous snarks and the new final pair that needs to be verified in a pairing check.
+/// For each previous snark, we concatenate all instances into a single vector. We return a vector of vectors,
+/// one vector per snark, for convenience.
+#[allow(clippy::type_complexity)]
+pub fn aggregate_hybrid<'a, PCS>(
```

Comment:
will the name be a bit confusing?  hybrid.

here we just expose more data fields and load some protocol as witness.  not related to "hybrid" here?

Reply:
Shall I change it to `aggregate_as_witness` ?